sandswitches/comms.py

```diff
@@ -18,7 +19,23 @@
             '-o', 'ServerAliveInterval=5']
 
 
-def get_sftp(hostname, port=22, user='root', keyfile=None, password=None):
+def get_pkey(keyfile, prompt=True, password=None):
+    """Get a private key references prompting for password by default.
+    """
+    try:
+        return paramiko.RSAKey.from_private_key_file(keyfile)
+    except paramiko.ssh_exception.PasswordRequiredException:
+        if not prompt and not password:
+            raise
+        if not password and prompt:
+            password = getpass("Password for {}: ".format(keyfile))
```

Comment:
Ahh, didn't notice that. That said, I suspect you'll be able to pass a password to a key that isn't password locked, and it'll still work.